tests/Fractions.Tests/FractionSpecs/ToDataType/Method_ToDataType.cs

```diff
@@ -143,6 +143,54 @@ public class When_converting_NaN : Spec {
+        Half.CreateChecked(Fraction.NaN).Should().Be(Half.NaN);
+        Invoking(() => BigInteger.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => Int128.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => UInt128.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => long.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => ulong.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => int.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => int.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => uint.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => nint.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => nint.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => UIntPtr.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => short.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => ushort.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => byte.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+        Invoking(() => sbyte.CreateChecked(Fraction.NaN)).Should().Throw<DivideByZeroException>();
+    }
+
+    [Test]
+    public void CreateSaturating_should_return_zero_or_NaN() {
+        double.CreateSaturating(Fraction.NaN).Should().Be(double.NaN);
+        Complex.CreateSaturating(Fraction.NaN).Should().Be(new Complex(double.NaN, 0));
+        decimal.CreateSaturating(Fraction.NaN).Should().Be(decimal.Zero);
+        float.CreateSaturating(Fraction.NaN).Should().Be(float.NaN);
+        Half.CreateSaturating(Fraction.NaN).Should().Be(Half.NaN);
+        BigInteger.CreateSaturating(Fraction.NaN).Should().Be(BigInteger.Zero);
+        Int128.CreateSaturating(Fraction.NaN).Should().Be(Int128.Zero);
+        UInt128.CreateSaturating(Fraction.NaN).Should().Be(UInt128.Zero);
+        long.CreateSaturating(Fraction.NaN).Should().Be(0);
+        ulong.CreateSaturating(Fraction.NaN).Should().Be(default);
+        int.CreateSaturating(Fraction.NaN).Should().Be(default);
+        uint.CreateSaturating(Fraction.NaN).Should().Be(default);
+        nint.CreateSaturating(Fraction.NaN).Should().Be(nint.Zero);
+        UIntPtr.CreateSaturating(Fraction.NaN).Should().Be(UIntPtr.Zero);
+        short.CreateSaturating(Fraction.NaN).Should().Be(default);
+        ushort.CreateSaturating(Fraction.NaN).Should().Be(default);
+        byte.CreateSaturating(Fraction.NaN).Should().Be(default);
+        sbyte.CreateSaturating(Fraction.NaN).Should().Be(default);
+    }

```

Comment:
Don't know what the reason is, but this is how this is implemented in BCL.. I first thought it was just in order to avoid throwing an exception, but later realized that `BigInteger.CreateSaturating(double.PositiveInfinity)` actually throws an `OverflowException`..